modules/core/04-channel/keeper/upgrade.go

```diff
@@ -1017,6 +1032,17 @@ func (k Keeper) WriteErrorReceipt(ctx sdk.Context, portID, channelID string, upg
 		panic(errorsmod.Wrapf(types.ErrInvalidUpgradeSequence, "error receipt sequence (%d) must be greater than existing error receipt sequence (%d)", errorReceiptToWrite.Sequence, existingErrorReceipt.Sequence))
 	}
 
+	// Ensure that no upgrade attempt exists for the same sequence we are
+	// writing an error receipt for. This could lead to divergent behaviour
+	// on the counterparty.
+	if channel.UpgradeSequence <= errorReceiptToWrite.Sequence {
```

Comment:
I think this is causing a couple of tests to fail because of [this](https://github.com/cosmos/ibc-go/blob/16082d6ebb2746c7040bd40a0c3c8ecba6388bee/modules/core/04-channel/keeper/upgrade.go#L58).